### PR TITLE
Added pyqt-builder-1.12.2

### PR DESCRIPTION
**Pull request by shuryanc**
Added pyqt-builder-1.12.2

### CODE REVIEW

File: components/python/PyQt-builder/Makefile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+COMPONENT_CLASSIFICATION=	Development/Python
+COMPONENT_LICENSE=      GPLv3
+COMPONENT_LICENSE_FILE= LICENSE-GPL3
+
+PYTHON_VERSIONS=	3.9
+
+include $(WS_MAKE_RULES)/common.mk
+
+# Auto-generated dependencies
+REQUIRED_PACKAGES += library/python/setuptools-39
```
Comment (AndWac) [inline components/python/PyQt-builder/Makefile:53]: You seem to have missed the dependency on
library/python/sip-39
(check the package's metadata, eg. setup.cfg and setup.py).

File: components/python/PyQt-builder/Makefile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH= sha256:f62bb688d70e0afd88c413a8d994bda824e6cebd12b612902d1945c5a67edcd7
+COMPONENT_ARCHIVE_URL=	$(call pypi_url)
+COMPONENT_FMRI=	library/python/pyqtbuilder
+COMPONENT_CLASSIFICATION=	Development/Python
+COMPONENT_LICENSE=      GPLv3
+COMPONENT_LICENSE_FILE= LICENSE-GPL3
+
+PYTHON_VERSIONS=	3.9
+
```
Comment (AndWac) [inline components/python/PyQt-builder/Makefile:45]: This looks like simple tool and not a framework. So we might only want to have a single package pyqt-builder and not multiple packages (pyqt-builder, pyqt-build-39).
If so, just use
PYTHON_VERSION= 3.9
here and omit the -pyver part on the manifest. You can look at out ansible package for an example.

File: components/python/PyQt-builder/PyQt-builder.p5m [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
```
Comment (AndWac) [inline components/python/PyQt-builder/PyQt-builder.p5m:24]: setup.py states that installation of this tool requires packaging and sip (>= 6.3 and < 7). Thus, you'll need to add runtime dependencies here:

depend type=require fmri=library/python/packaging-39
depend type=require fmri=library/python/sip-39

File: components/python/PyQt-builder/PyQt-builder.p5m [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+depend type=require fmri=library/python/packaging-39
+depend type=require fmri=library/python/sip
+
```
Comment (AndWac) [inline components/python/PyQt-builder/PyQt-builder.p5m:27]: This should be sip-39 because this package is being build with python-3.9.

Comment (shuryanc) [reply components/python/PyQt-builder/PyQt-builder.p5m:27]: I think sip is a tool (sip has some executable files under /usr/bin), so I changed it to sip, how do you think?

Comment (AndWac) [reply components/python/PyQt-builder/PyQt-builder.p5m:27]: That would be ok then. Did you check that sip works with other python versions, too?
We will probably drop python 3.5 as soon as IPS works with 3.9. Probably 3.7 gets dropped next and 3.10 or 3.11 added.

Comment (shuryanc) [reply components/python/PyQt-builder/PyQt-builder.p5m:27]: Yes, tested building pyqt-builder by python3.7 and the 3.9 sip still works. I will try to make them to adapt new version of python when higher version is shipped.

**Pull request merged**